src/components/search/search.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import useSWRInfinite from 'swr/infinite';
+
+import {
+    Button,
+    TextInput,
+    Select,
+    ESystemIconNames,
+    TypographyScale,
+} from '@mdb/flora';
+
+import { ContentPiece } from '../../interfaces/content-piece';
+
+import Results from './results';
+import {
+    titleStyles,
+    searchBoxStyles,
+    sortBoxStyles,
+    loadMoreStyles,
+} from './styles';
+import { SearchProps, SortByType } from './types';
+import { fetcher, sortByOptions } from './utils';
+
+const Search: React.FunctionComponent<SearchProps> = ({
+    name,
+    hideSortBy = false,
+    filters = [],
+}) => {
+    const [search, setSearch] = useState('');
+    const [sortBy, setSortBy] = useState<SortByType>('recent');
+
+    const getKey = (pageIndex: number, previousPageData: ContentPiece[]) => {
+        if (previousPageData && !previousPageData.length) return null;
+        return `?search=${search}&sort=${sortBy}&page=${pageIndex}&filters=${filters.join(
+            ','
+        )}`;
+    };
+    const { data, size, setSize, error, isValidating } = useSWRInfinite(
+        getKey,
+        fetcher,
+        {
+            revalidateIfStale: false,
+            revalidateOnFocus: false,
+            revalidateOnReconnect: false,
+            shouldRetryOnError: false,
+        }
+    );
+
+    return (
+        <>
+            <TypographyScale variant="heading5" sx={titleStyles}>
+                All {name} Content
+            </TypographyScale>
+            <div sx={searchBoxStyles}>
+                <TextInput
+                    name="search-box"
+                    label={`Search ${name} Content`}
+                    iconName={ESystemIconNames.SEARCH}
+                    value={search}
+                    onChange={val => {
```

Comment:
nitpick but you can just swap `val` with either `e` or `event` since you already have to specifying that you're getting a value with the `value` property off of the `target` property. Hope this makes sense.

Reply:
In fact not a biggie but for performance and not creating a new function object on every typing of character the user does you can create a function like this:

```
const onSearch = (e) => {
        setSize(1);
        setSearch(val.target.value);
}
```

Then on the onChange prop just set it like so:

`onChange={onSearch}`

This way on every character update you're referencing the named function `onSearch` in memory versus creating a new function object. Hope this makes sense.

Reply:
Ah ok that makes sense, but does that still apply if we are re-rendering the component whenever we type? I'll still make the change to be on the safe side, but just curious how react handles function creation on render.